semanticdb/semanticdb/semanticdb.proto

```diff
@@ -320,5 +320,55 @@ message Diagnostic {
 
 message Synthetic {
   Range range = 1;
-  TextDocument text = 2;
+  Tree tree = 2;
+}
+
+message Tree {
+  oneof sealed_value {
+    ApplyTree apply = 1;
+    FunctionTree function = 2;
+    IdTree id = 3;
+    LiteralTree literal = 4;
+    MacroExpansionTree macroExpansion = 5;
+    OriginalTree original = 6;
+    SelectTree select = 7;
+    TypeApplyTree typeApply = 8;
+  }
+}
+
+message ApplyTree {
+  Tree fn = 1;
+  repeated Tree args = 2;
+}
+
+message FunctionTree {
+  repeated IdTree params = 1;
+  Tree term = 2;
+}
+
+message IdTree {
+  string sym = 1;
+}
+
+message LiteralTree {
+  Constant const = 1;
+}
+
+message MacroExpansionTree {
+  Tree expandee = 1;
```

Comment:
Expandee can be an arbitrary term, so this is not going to work unless we model all Scalac terms. I suggest we just drop this field.